### PR TITLE
[B-894] Investigate terraform

### PR DESCRIPTION
**Pull request by Bernardstanislas**
So far I:
- mapped all the apps
- mapped all the pipelines
- mapped all the resources

Left to do, in another PR:
- map all the configs
- map all the drains
- map all the domains

### CODE REVIEW

File: provisioning/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+Review the output, check that the modifications planned are correct. If so:
+
+```
+terraform apply -var-file {develop,staging,production}.tfvars
+```
+
+## Importing existing infrastructure under Terraform supervision
+
+All the infrastructure already exists on Heroku.
```
Comment (AnthonyBuisset) [inline provisioning/README.md:40]: Excellent, this answers a question I had 😊

File: provisioning/heroku_app_module/main.tf [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  stage    = var.stage
+}
+
+resource "heroku_addon" "database" {
+  count = var.database_billing_app ? 1 : 0
+  app   = heroku_app.app.name
+  plan  = var.postgres_plan
+}
+
+resource "heroku_addon" "queue" {
```
Comment (AnthonyBuisset) [inline provisioning/heroku_app_module/main.tf:22]: Could we name it `amqp` to match our infrastructure module in the code ?

**Review by AnthonyBuisset — APPROVED** [review]

**Comment by AnthonyBuisset** [comment]
> So far I:
> - mapped all the apps
> - mapped all the pipelines
> - mapped all the resources
> 
> Left to do, in another PR:
> - map all the configs
> - map all the drains
> - map all the domains

I guess we also need to map the Dyno as well ?

**Comment by Bernardstanislas** [comment]
> > So far I:
> > 
> > * mapped all the apps
> > * mapped all the pipelines
> > * mapped all the resources
> > 
> > Left to do, in another PR:
> > 
> > * map all the configs
> > * map all the drains
> > * map all the domains
> 
> I guess we also need to map the Dyno as well ?

oh right, there is also a module for that: https://registry.terraform.io/providers/heroku/heroku/latest/docs/resources/formation

File: provisioning/heroku_app_module/main.tf (hunk 2 above)
Comment (Bernardstanislas) [reply provisioning/heroku_app_module/main.tf:22]: yes!

**Pull request merged**